climakitae/data_loaders.py

```diff
@@ -80,21 +80,8 @@ def _open_and_concat(file_list, selections, cat, ds_region):
             )
         )
         # subset data spatially:
-        data_crs = ccrs.CRS(pyproj.CRS.from_cf(data['Lambert_Conformal'].attrs))
-
         if ds_region:
-            output = data_crs.transform_points(ccrs.PlateCarree(),
-                                                   x=ds_region.coords[0][:,0],
-                                                   y=ds_region.coords[0][:,1])
-
-            data = data.sel(x=slice(np.nanmin(output[:,0]), np.nanmax(output[:,0])),
-                y=slice(np.nanmin(output[:,1]), np.nanmax(output[:,1])))
-
-            mask = ds_region.mask(data.lon, data.lat, wrap_lon=False)
-            assert (
-                False in mask.isnull()
-            ), "Insufficient gridcells are contained within the bounds."
-            data = data.where(np.isnan(mask) == False)
+            data = data.rio.clip(geometries=ds_region, crs=4326, drop=True)
```

Comment:
This seems to be the only needed operation and it runs as fast as all this other code.